src/renderer/components/dialog/OpdsFeedAddForm.tsx

```diff
@@ -20,54 +20,96 @@ interface Props extends TranslatorProps {
 interface State {
     name: string;
     url: string;
+    opdsFeedList: {
+        [name: string]: string;
+    } | undefined;
+    selectValue: string;
 }
 
 export class OpdsFeedAddForm extends React.Component<Props, State> {
-    constructor(props: any) {
+    private defaultSelectValue = "undefined";
+    private opdsFeedListServerUrl = "http://thorium-opds.edrlab.org/";
+
+    constructor(props: Props) {
         super(props);
 
         this.state = {
-            name: undefined,
-            url: undefined,
+            name: "",
+            url: "",
+            opdsFeedList: undefined,
+            selectValue: this.defaultSelectValue,
         };
 
         this.add = this.add.bind(this);
         this.close = this.close.bind(this);
+        this.onSelectOpdsFeedChange = this.onSelectOpdsFeedChange.bind(this);
+    }
+
+    public async componentDidMount() {
+        try {
+            const response = await fetch(this.opdsFeedListServerUrl);
```

Comment:
Note that usage of the `window.fetch` Web API will probably be affected by the activation of `webSecurity` in the `BrowserWindow`. Possible workarounds:
1) ensure adequate CORS headers are added to HTTP responses (server side)
2) because `nodeIntegration` is enabled, switch from `window.fetch` to a NodeJS `request` API

More information:
https://github.com/readium/readium-desktop/issues/736

Reply:
I think the CORS on server is enable, I activated it

Reply:
I see `Access-Control-Allow-Origin: *` in the HTTP response, so it should be okay.